### PR TITLE
Replace setproctitle dependency

### PR DESCRIPTION
**Pull request by markypizz**
[Edited]

**Description**

Replace `setproctitle` dependency with our own small helper

**Verification**

Passing CI, observed proc title change when running `modeld`, (see https://github.com/commaai/openpilot/pull/32666#issuecomment-2156242684)

### CODE REVIEW

File: common/setproctitle.py [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import os
+from openpilot.common.run import run_cmd
+
+def getproctitle():
+    pid = os.getpid()
```
Comment (adeebshihadeh) [inline common/setproctitle.py:5]: we use two space indentation

File: common/setproctitle.py [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+    proctitle = run_cmd([f"cat /proc/{pid}/comm"])
```
Comment (adeebshihadeh) [inline common/setproctitle.py:6]: make sure to check if we're on linux

**Review by adeebshihadeh — CHANGES_REQUESTED** [review]
Welcome, and yeah this is the right general direction.

Tests are failing because there's still a `setproctitle` import. CI green check mark + verify the title shows up in `htop` is enough validation for this change

**Comment by markypizz** [comment]
Addresses https://github.com/commaai/openpilot/issues/32660 . Let me know of anything else to change. Thanks!

```openpilot-py3.11vscode ➜ /workspaces/ubuntu $ ps -a
    PID TTY          TIME CMD
  37224 pts/0    00:00:01 poetry
  49515 pts/1    00:00:15 loggerd
  49533 pts/1    00:00:15 loggerd
  95648 pts/1    00:00:00 python3
 103095 pts/1    00:00:00 selfdrive.model
 103204 pts/7    00:00:00 ps```

**Comment by adeebshihadeh** [comment]
Doesn't seem to work with `htop` still.

**Comment by adeebshihadeh** [comment]
Ended up merging #32667 since it made the agnos-builder PR and had a test.

**Pull request closed without merging**